src/xtgeo/grid3d/_grid_import.py

```diff
@@ -49,6 +50,13 @@ def from_file(
         result.update(_grid_import_xtgcpgeom.import_xtgcpgeom(gfile, **kwargs))
     elif fformat == FileFormat.HDF:
         result.update(_grid_import_xtgcpgeom.import_hdf5_cpgeom(gfile, **kwargs))
+    else:
+        raise InvalidFileFormatError(
+            f"File format {fformat} is invalid for type Grid. Supported formats are "
+            "'roff', 'roff_binary', 'roff_ascii', "
+            "'egrid', 'fegrid', 'grdecl', 'bgrdecl'"
+            "'hdf', 'xtg'."
+        )
```

Comment:
I guess we have the enum with `FileFormat`, there is probably a generic way to put them into a list with some method on it. I'll see if it's reasonable.

I also made #1164 to make some things around this a bit easier but it could be useful for that as well.